store/modules/actions/authCore.js

```diff
@@ -127,14 +127,19 @@ export async function fetchAuthCoreCosmosWallet({ state }) {
 }
 
 export async function prepareAuthCoreCosmosTxSigner({ state }) {
+  let cosmosProvider;
   if (!state.cosmosProvider) throw new Error('COSMOS_WALLET_NOT_INITED');
-  return async function signer(signMessage) {
-    const data = JSON.parse(signMessage);
-    const dataWithSign = await state.cosmosProvider.sign(data);
-    const signObject = dataWithSign.signatures[dataWithSign.signatures.length - 1];
+  if (state.cosmosProvider) {
+    cosmosProvider = state.cosmosProvider; // eslint-disable-line prefer-destructuring
+  }
```

Comment:
since we will throw `COSMOS_WALLET_NOT_INITED` above, `state.cosmosProvider` always exists and not need for checking